File: PROYECTO/V1/Zarpas_Backend/src/main/java/com/gestion/zarpas_backend/modelo/Publicacion.java
package com.gestion.zarpas_backend.modelo;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;

import java.sql.Timestamp;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "publicacion")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Publicacion {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_publicacion")
    private Long idPublicacion;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "id_usuario", nullable = false)
    @JsonBackReference("usuario-publicaciones")
    private Usuario usuario;

    private String titulo;

    @Column(columnDefinition = "TEXT")
    private String contenido;

    @Column(name = "imagen_url")
    private String imagenUrl;

    @Column(name = "fecha_creacion")
    private Timestamp fechaCreacion;

    @Column(name = "fecha_modificacion")
    private Timestamp fechaModificacion;

    @OneToMany(mappedBy = "publicacion", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference("publicacion-comentarios")
    private List<Comentario> comentarios = new ArrayList<>();

    @OneToMany(mappedBy = "publicacion", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference("publicacion-publicacionGuardadas")
    private List<PublicacionGuardada> guardadosPorUsuarios = new ArrayList<>();

    @OneToMany(mappedBy = "publicacion", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference("publicacion-reacciones")
    private List<ReaccionPublicacion> reaccionesPublicacion = new ArrayList<>();

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "id_categoria", nullable = false)
    private Categoria categoria;
}

File: PROYECTO/V1/Zarpas_Frontend/gestion-zarpas-frontend/src/app/modelos/Publicacion.ts
import { Comentario } from "./Comentario";
import { Usuario } from "./Usuario";
import { TipoReaccion } from "./TipoReaccion";
import { ReaccionPublicacion } from "./ReaccionPublicacion";
import { Categoria } from "./Categoria";
import { PublicacionGuardada } from './PublicacionGuardada';

export interface Publicacion {
    idPublicacion: number;
    usuario: Usuario;
    titulo: string;
    contenido: string;
    imagenUrl?: string | null;
    fechaCreacion: Date | null;
  fechaModificacion?: Date | null;
    comentarios?: Comentario[];
    usuariosGuardaron?: Usuario[];
    reaccionesPublicacion?: ReaccionPublicacion[];
    conteoLikes?: number;
    conteoDislikes?: number;
    miReaccion?: TipoReaccion | null;
    categoria?: Categoria | null;

    mostrarComentarios?: boolean; 
    guardadosPorUsuarios?: PublicacionGuardada[]; 
    isSaved?: boolean; 
  }